server/__init__.py

```diff
@@ -17,32 +17,37 @@
 #  limitations under the License.
 ###############################################################################
 
-from girder import events
-from girder.api import access
-from girder.api.rest import boundHandler
+from bson import json_util
+import json
+
 from girder.constants import SettingKey
+from girder.models.folder import Folder
+from girder.models.item import Item
 from girder.models.setting import Setting
 
 from . import rest
 
 
-@access.public
-@boundHandler
-def _virtualChildItems(self, event):
-    params = event.info['params']
+originalChildItems = None
+
 
-    if params['type'] != 'folder':
-        return  # This can't be a virtual folder
-    try:
-        import girder.plugins.virtual_folders
-    except ImportError:
-        return  # If virtual folders aren't enabled, do nothing
-    params['folderId'] = event.info['id']
-    return girder.plugins.virtual_folders._virtualChildItems(event)
+def childItems(self, folder, limit=0, offset=0, sort=None, filters=None,
+               includeVirtual=False, **kwargs):
+    if not includeVirtual or not folder.get('isVirtual') or 'virtualItemsQuery' not in folder:
+        return originalChildItems(self, folder, limit=limit, offset=offset,
+                                  sort=sort, filters=filters, **kwargs)
+    q = json_util.loads(folder['virtualItemsQuery'])
+    if 'virtualItemsSort' in folder and sort is None:
+        sort = json.loads(folder['virtualItemsSort'])
+    q.update(filters or {})
```

Comment:
wouldn't it be better to combine them using a logical and?

as in
```python
if filters:
  q = {
    '$and': [q, filters ]
  }
```

Reply:
This mimics the code that is in girder core (https://github.com/girder/girder/blob/master/girder/models/folder.py#L410).  At least there, the intent seems to be able to replace part of the query expression.  Perhaps girder core should change, but I'd rather have this be the same as what is there.

Reply:
see https://github.com/girder/girder/issues/3130

Reply:
since the issue has been closed, I will resolve this issue to be complaint with the rest of girder